cluster-api/src/test/java/io/aiven/klaw/clusterapi/services/MonitoringServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package io.aiven.klaw.clusterapi.services;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import io.aiven.klaw.clusterapi.models.OffsetDetails;
+import io.aiven.klaw.clusterapi.models.enums.KafkaSupportedProtocol;
+import io.aiven.klaw.clusterapi.utils.ClusterApiUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeTopicsResult;
+import org.apache.kafka.clients.admin.ListOffsetsResult;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MonitoringServiceTest {
+
+  @Mock private ClusterApiUtils clusterApiUtils;
+  @Mock private KafkaSupportedProtocol protocol;
+  @Mock private AdminClient adminClient;
+  @Mock private DescribeTopicsResult describeTopicsResult;
+  @Mock private TopicPartitionInfo topicPartitionInfo;
+  @Mock private ListOffsetsResult listOffsetsEarliestResult;
+  private MonitoringService monitoringService;
+
+  @BeforeEach
+  void setUp() {
+    monitoringService = new MonitoringService(clusterApiUtils);
+  }
+
+  @Test
+  void getConsumerGroupDetails() throws Exception {
+    String consumerGroupId = "consumerGroupId";
+    String topicName = "topicName";
+    String environment = "environment";
+    String clusterName = "clusterName";
+    int partition = 1;
```

Comment:
If you can make these class constants as well 
 String consumerGroupId = "consumerGroupId";
    String topicName = "topicName";
    String environment = "environment";
    String clusterName = "clusterName";

Reply:
Adding to new TestConstants class